### PR TITLE
Silicon Accents

### PR DESCRIPTION
**Pull request by Geevies**
* Stationbounds now have accents, starting off with Text-to-Speech, and can choose to mimic any accent.
* The bots that robotics can produce now also have TTS accents.
* IPCs can choose to use the TTS accent if they wish.

### CODE REVIEW

**Comment by Geevies** [comment]
!review

File: code/modules/mob/hear_say.dm [hunk 1]
```diff
@@ -48,7 +48,9 @@
 		var/mob/living/carbon/human/H = speaker
 		speaker_name = H.GetVoice()
 
-		accent_icon = H.get_accent_icon(language)
+	if(isliving(speaker))
+		var/mob/living/L = speaker
+		accent_icon = L.get_accent_icon(language)
```
Comment (NonQueueingMatt) [inline code/modules/mob/hear_say.dm:53]: Uhhh... you already cast this as mob/living/carbon/human, why are you then checking if it's living and then reassigning it?

File: code/modules/mob/living/living.dm [hunk 2]
```diff
@@ -958,3 +958,9 @@ default behaviour is:
 
 /mob/living/proc/needs_wheelchair()
 	return FALSE
+
+/mob/living/proc/get_accent_icon(var/datum/language/speaking = null)
+	if(accent && speaking && speaking.allow_accents)
+		var/datum/accent/a = SSrecords.accents[accent]
+		var/final_icon = a.tag_icon
+		return "<IMG src='\ref['./icons/accent_tags.dmi']' class='text_tag' iconstate='[final_icon]'>"
```
Comment (NonQueueingMatt) [inline code/modules/mob/living/living.dm:966]: You can avoid copypaste in the human proc by making it do ..() at the  end of it, I think

File: code/modules/mob/living/living_defines.dm [hunk 3]
```diff
@@ -2,6 +2,8 @@
 	see_in_dark = 2
 	see_invisible = SEE_INVISIBLE_LIVING
 
+	var/accent = ""
```
Comment (NonQueueingMatt) [inline code/modules/mob/living/living_defines.dm:5]:
```suggestion
	var/accent
```
You don't need to initialize strings unless you plan on doing +=

**Comment by Chada-1** [comment]
THIS IS COOL

File: code/modules/mob/mob_helpers.dm [hunk 4]
```diff
@@ -1224,4 +1224,11 @@ proc/is_blind(A)
 		W.add_fingerprint(src)
 
 /mob/proc/set_intent(var/set_intent)
-	a_intent = set_intent
\ No newline at end of file
+	a_intent = set_intent
+
+/mob/proc/get_accent_icon(var/datum/language/speaking = null, var/force_accent)
```
Comment (skull132) [inline code/modules/mob/mob_helpers.dm:1229]: This needs a `SHOULD_CALL_PARENT(TRUE)` define.

**Pull request merged**